Newly created plugin project fails on tests

Hello,

I just tried to create my own plugin using this project. However, when running `npm test` two of the tests fail:

````
  PreProcess Snapshot testing
PreProcess: config=<true>
    √ Test petstore
PreProcess: config=<true>
    √ Test two schema pattern

  PostProcess Snapshot testing
PostProcess: config=<true>
    1) Test complex patterns
PostProcess: config=<true>
    2) Test simple example


  2 passing (162ms)
  2 failing

  1) PostProcess Snapshot testing
       Test complex patterns:

      AssertionError [ERR_ASSERTION]: 
C:\Users\Chris\Documents\dtsgen-file\dtsgenerator-multi-file\test\post_snapshots\complex-patterns
declare namespace JsonSchemaOrg {
    namespace Complex {
        /**
         * complex patterns
         */
        export interface Patterns {
            p: Patterns.Definitions.Primitives;
            pr: Patterns.Definitions.PrimitivesRequired;
            ut?: Patterns.Definitions.UnionTuple;
            at?: Patterns.Definitions.ArrayTypes;
            nt?: Patterns.Definitions.NestedTypes;
            ct?: Patterns.Definitions.CommentTest;
            ta?: Patterns.Definitions.TypeAlias;
            array?: Patterns.Definitions.TypeArray;
            const?: Patterns.Definitions.MathPi | Patterns.Definitions.IsDebug | Patterns.Definitions.IsTest | Patterns.Definitions.ProjectName;
        }
        namespace Patterns {
            namespace Definitions {
                export interface ArrayTypes {
                    strings?: string[];
                    numbers?: number[];
                    arrays?: string[][][];
                    array_of_array?: (string | string[])[];
                }
                /**
                 * comment test type.
                 * description comment.
                 * example:
                 *   obj = {
                 *     p1: 'example',
                 *     p2: true,
                 *     p3: [ false, 1.23, 'tuple' ],
                 *   }
                 */
                export interface CommentTest {
                    /**
                     * p1 is string type.
                     */
                    p1: string;
                    /**
                     * p2 is union types.
                     * example:
                     * true or 1 or 'string'
                     */
                    p2: boolean | string | number;
                    /**
                     * p3 is tuple types
                     * example:
                     * true
                     * 2.5
                     * p3
                     */
                    p3: [boolean, number, string?, ...any[]];
                }
                export type IsDebug = false;
                export type IsTest = true;
                export type MathPi = 3.1415926536;
                export interface NestedTypes {
                    first: {
                        second: {
                            third: {};
                        };
                    };
                }
                export interface Primitives {
                    readonly any?: any;
                    array?: any[];
                    boolean?: boolean;
                    double?: number; // double
                    int?: number; // int
                    integer?: number;
                    null?: null;
                    number?: number;
                    object?: {};
                    string?: string;
                    undefined?: undefined;
                }
                export interface PrimitivesRequired {
                    readonly any: any;
                    array: any[];
                    boolean: boolean;
                    double: number; // double
                    int: number; // int
                    integer: number;
                    null: null;
                    number: number;
                    object: {};
                    string: string;
                    undefined: undefined;
                }
                export type ProjectName = "dtsgenerator";
                export type TypeAlias = Primitives | PrimitivesRequired;
                export type TypeArray = {
                    a: string;
                    b?: string;
                    n?: {
                        c: number;
                    }[];
                }[];
                export interface UnionTuple {
                    s_tuple?: "A" | "B" | "C";
                    n_tuple?: 1 | 2 | 3 | 4 | 5;
                    some_types?: boolean | number | string;
                    ref_types?: Primitives | PrimitivesRequired;
                }
            }
        }
    }
}


      + expected - actual

      -declare namespace JsonSchemaOrg {
      -    namespace Complex {
      -        /**
      -         * complex patterns
      -         */
      -        export interface Patterns {
      -            p: Patterns.Definitions.Primitives;
      -            pr: Patterns.Definitions.PrimitivesRequired;
      -            ut?: Patterns.Definitions.UnionTuple;
      -            at?: Patterns.Definitions.ArrayTypes;
      -            nt?: Patterns.Definitions.NestedTypes;
      -            ct?: Patterns.Definitions.CommentTest;
      -            ta?: Patterns.Definitions.TypeAlias;
      -            array?: Patterns.Definitions.TypeArray;
      -            const?: Patterns.Definitions.MathPi | Patterns.Definitions.IsDebug | Patterns.Definitions.IsTest | Patterns.Definitions.ProjectName;
      -        }
      -        namespace Patterns {
      -            namespace Definitions {
      -                export interface ArrayTypes {
      -                    strings?: string[];
      -                    numbers?: number[];
      -                    arrays?: string[][][];
      -                    array_of_array?: (string | string[])[];
      -                }
      -                /**
      -                 * comment test type.
      -                 * description comment.
      -                 * example:
      -                 *   obj = {
      -                 *     p1: 'example',
      -                 *     p2: true,
      -                 *     p3: [ false, 1.23, 'tuple' ],
      -                 *   }
      -                 */
      -                export interface CommentTest {
      -                    /**
      -                     * p1 is string type.
      -                     */
      -                    p1: string;
      -                    /**
      -                     * p2 is union types.
      -                     * example:
      -                     * true or 1 or 'string'
      -                     */
      -                    p2: boolean | string | number;
      -                    /**
      -                     * p3 is tuple types
      -                     * example:
      -                     * true
      -                     * 2.5
      -                     * p3
      -                     */
      -                    p3: [boolean, number, string?, ...any[]];
      -                }
      -                export type IsDebug = false;
      -                export type IsTest = true;
      -                export type MathPi = 3.1415926536;
      -                export interface NestedTypes {
      -                    first: {
      -                        second: {
      -                            third: {};
      -                        };
      -                    };
      -                }
      -                export interface Primitives {
      -                    readonly any?: any;
      -                    array?: any[];
      -                    boolean?: boolean;
      -                    double?: number; // double
      -                    int?: number; // int
      -                    integer?: number;
      -                    null?: null;
      -                    number?: number;
      -                    object?: {};
      -                    string?: string;
      -                    undefined?: undefined;
      -                }
      -                export interface PrimitivesRequired {
      -                    readonly any: any;
      -                    array: any[];
      -                    boolean: boolean;
      -                    double: number; // double
      -                    int: number; // int
      -                    integer: number;
      -                    null: null;
      -                    number: number;
      -                    object: {};
      -                    string: string;
      -                    undefined: undefined;
      -                }
      -                export type ProjectName = "dtsgenerator";
      -                export type TypeAlias = Primitives | PrimitivesRequired;
      -                export type TypeArray = {
      -                    a: string;
      -                    b?: string;
      -                    n?: {
      -                        c: number;
      -                    }[];
      -                }[];
      -                export interface UnionTuple {
      -                    s_tuple?: "A" | "B" | "C";
      -                    n_tuple?: 1 | 2 | 3 | 4 | 5;
      -                    some_types?: boolean | number | string;
      -                    ref_types?: Primitives | PrimitivesRequired;
      -                }
      -            }
      -        }
      -    }
      -}
      +declare namespace JsonSchemaOrg {
      +    namespace Complex {
      +        /**
      +         * complex patterns
      +         */
      +        export interface Patterns {
      +            p: Patterns.Definitions.Primitives;
      +            pr: Patterns.Definitions.PrimitivesRequired;
      +            ut?: Patterns.Definitions.UnionTuple;
      +            at?: Patterns.Definitions.ArrayTypes;
      +            nt?: Patterns.Definitions.NestedTypes;
      +            ct?: Patterns.Definitions.CommentTest;
      +            ta?: Patterns.Definitions.TypeAlias;
      +            array?: Patterns.Definitions.TypeArray;
      +            const?: Patterns.Definitions.MathPi | Patterns.Definitions.IsDebug | Patterns.Definitions.IsTest | Patterns.Definitions.ProjectName;
      +        }
      +        namespace Patterns {
      +            namespace Definitions {
      +                export interface ArrayTypes {
      +                    strings?: string[];
      +                    numbers?: number[];
      +                    arrays?: string[][][];
      +                    array_of_array?: (string | string[])[];
      +                }
      +                /**
      +                 * comment test type.
      +                 * description comment.
      +                 * example:
      +                 *   obj = {
      +                 *     p1: 'example',
      +                 *     p2: true,
      +                 *     p3: [ false, 1.23, 'tuple' ],
      +                 *   }
      +                 */
      +                export interface CommentTest {
      +                    /**
      +                     * p1 is string type.
      +                     */
      +                    p1: string;
      +                    /**
      +                     * p2 is union types.
      +                     * example:
      +                     * true or 1 or 'string'
      +                     */
      +                    p2: boolean | string | number;
      +                    /**
      +                     * p3 is tuple types
      +                     * example:
      +                     * true
      +                     * 2.5
      +                     * p3
      +                     */
      +                    p3: [boolean, number, string?, ...any[]];
      +                }
      +                export type IsDebug = false;
      +                export type IsTest = true;
      +                export type MathPi = 3.1415926536;
      +                export interface NestedTypes {
      +                    first: {
      +                        second: {
      +                            third: {};
      +                        };
      +                    };
      +                }
      +                export interface Primitives {
      +                    readonly any?: any;
      +                    array?: any[];
      +                    boolean?: boolean;
      +                    double?: number; // double
      +                    int?: number; // int
      +                    integer?: number;
      +                    null?: null;
      +                    number?: number;
      +                    object?: {};
      +                    string?: string;
      +                    undefined?: undefined;
      +                }
      +                export interface PrimitivesRequired {
      +                    readonly any: any;
      +                    array: any[];
      +                    boolean: boolean;
      +                    double: number; // double
      +                    int: number; // int
      +                    integer: number;
      +                    null: null;
      +                    number: number;
      +                    object: {};
      +                    string: string;
      +                    undefined: undefined;
      +                }
      +                export type ProjectName = "dtsgenerator";
      +                export type TypeAlias = Primitives | PrimitivesRequired;
      +                export type TypeArray = {
      +                    a: string;
      +                    b?: string;
      +                    n?: {
      +                        c: number;
      +                    }[];
      +                }[];
      +                export interface UnionTuple {
      +                    s_tuple?: "A" | "B" | "C";
      +                    n_tuple?: 1 | 2 | 3 | 4 | 5;
      +                    some_types?: boolean | number | string;
      +                    ref_types?: Primitives | PrimitivesRequired;
      +                }
      +            }
      +        }
      +    }
      +}

      at Context.<anonymous> (test\snapshot_test.ts:133:24)
      at step (node_modules\tslib\tslib.js:140:27)
      at Object.next (node_modules\tslib\tslib.js:121:57)
      at fulfilled (node_modules\tslib\tslib.js:111:62)

  2) PostProcess Snapshot testing
       Test simple example:

      AssertionError [ERR_ASSERTION]: 
C:\Users\Chris\Documents\dtsgen-file\dtsgenerator-multi-file\test\post_snapshots\simple-example
declare namespace JsonSchemaOrg {
    /**
     * A geographical coordinate
     */
    export interface Geo {
        latitude?: number;
        longitude?: number;
    }
    /**
     * Product set
     */
    export type SimpleExample = {
        /**
         * The unique identifier for a product
         */
        id: number;
        name: string;
        price: number;
        tags?: string[];
        dimensions?: {
            length: number;
            width: number;
            height: number;
        };
        /**
         * Coordinates of the warehouse with the product
         */
        warehouseLocation?: Geo;
    }[];
}


      + expected - actual

      -declare namespace JsonSchemaOrg {
      -    /**
      -     * A geographical coordinate
      -     */
      -    export interface Geo {
      -        latitude?: number;
      -        longitude?: number;
      -    }
      -    /**
      -     * Product set
      -     */
      -    export type SimpleExample = {
      -        /**
      -         * The unique identifier for a product
      -         */
      -        id: number;
      -        name: string;
      -        price: number;
      -        tags?: string[];
      -        dimensions?: {
      -            length: number;
      -            width: number;
      -            height: number;
      -        };
      -        /**
      -         * Coordinates of the warehouse with the product
      -         */
      -        warehouseLocation?: Geo;
      -    }[];
      -}
      +declare namespace JsonSchemaOrg {
      +    /**
      +     * A geographical coordinate
      +     */
      +    export interface Geo {
      +        latitude?: number;
      +        longitude?: number;
      +    }
      +    /**
      +     * Product set
      +     */
      +    export type SimpleExample = {
      +        /**
      +         * The unique identifier for a product
      +         */
      +        id: number;
      +        name: string;
      +        price: number;
      +        tags?: string[];
      +        dimensions?: {
      +            length: number;
      +            width: number;
      +            height: number;
      +        };
      +        /**
      +         * Coordinates of the warehouse with the product
      +         */
      +        warehouseLocation?: Geo;
      +    }[];
      +}

      at Context.<anonymous> (test\snapshot_test.ts:133:24)
      at step (node_modules\tslib\tslib.js:140:27)
      at Object.next (node_modules\tslib\tslib.js:121:57)
      at fulfilled (node_modules\tslib\tslib.js:111:62)



npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! dtsgenerator-multi-file@0.1.0 do-test: `cross-env TS_NODE_FILES=true mocha --exit --require ts-node/register --colors test/*_test.ts`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the dtsgenerator-multi-file@0.1.0 do-test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Chris\AppData\Roaming\npm-cache\_logs\2020-07-13T18_31_53_725Z-debug.log
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 | 
 index.ts |     100 |      100 |     100 |     100 | 
----------|---------|----------|---------|---------|-------------------

=============================== Coverage summary ===============================
Statements   : 100% ( 10/10 )
Branches     : 100% ( 0/0 )
Functions    : 100% ( 5/5 )
Lines        : 100% ( 9/9 )
================================================================================
npm ERR! Test failed.  See above for more details.
````

In case that makes a difference: I am running Windows 10. 

Thanks a lot,
Christian

--- a/template/test/snapshot_test.ts
+++ b/template/test/snapshot_test.ts
@@ -7,7 +7,7 @@ import plugin from '..';
 import assert = require('assert');
 
 const splitStringByNewLine = (input: string): string[] => {
-    const splitted = input.match(/[^\r\n]+/g)
+    const splitted = input.split(/\r?\n/);
     return splitted ? splitted : [];
 } 
 

--- a/template/test/snapshot_test.ts
+++ b/template/test/snapshot_test.ts
@@ -6,6 +6,11 @@ import plugin from '..';
 
 import assert = require('assert');
 
+const splitStringByNewLine = (input: string): string[] => {
+    const splitted = input.match(/[^\r\n]+/g)
+    return splitted ? splitted : [];
+} 
+
 describe('PreProcess Snapshot testing', () => {
     const fixturesDir = path.join(__dirname, 'pre_snapshots');
     const inputFileName = 'input.json';
@@ -59,9 +64,9 @@ describe('PreProcess Snapshot testing', () => {
                 const expected = fs.readFileSync(expectedFilePath, {
                     encoding: 'utf-8',
                 });
-                assert.equal(
-                    actual,
-                    expected,
+                assert.deepEqual(
+                    splitStringByNewLine(actual),
+                    splitStringByNewLine(expected),
                     `
 ${fixtureDir}
 ${actual}
@@ -130,9 +135,9 @@ describe('PostProcess Snapshot testing', () => {
                 const expected = fs.readFileSync(expectedFilePath, {
                     encoding: 'utf-8',
                 });
-                assert.equal(
-                    actual,
-                    expected,
+                assert.deepEqual(
+                   splitStringByNewLine(actual),
+                   splitStringByNewLine(expected),
                     `
 ${fixtureDir}
 ${actual}

